File: Python/122_Best_Time_to_Buy_and_Sell_Stock_II.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        profit = 0
        for i in range(len(prices)-1):
            gap = prices[i+1] - prices[i]
            if gap > 0:
                profit += gap
        return profit

File: Cplusplus/122_Best_Time_to_Buy_and_Sell_Stock_II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        if (n<=1) return 0;
        
        int profit = 0;
        int gap = 0;
        for (int i=0; i<n-1; i++) {
            gap = prices[i+1] - prices[i];
            if (gap > 0) {
                profit += gap;
            }
        }
        return profit;
    }
};
